protocols/grp/adjRIBIn/adj_rib_in.go

```diff
@@ -0,0 +1,254 @@
+package adjRIBIn
+
+import (
+	"sync"
+
+	"github.com/bio-routing/bio-rd/net"
+	"github.com/bio-routing/bio-rd/route"
+	"github.com/bio-routing/bio-rd/routingtable"
+	"github.com/bio-routing/bio-rd/routingtable/filter"
+	"github.com/bio-routing/bio-rd/routingtable/vrf"
+	"github.com/bio-routing/bio-rd/util/log"
+)
+
+// AdjRIBIn represents an Adjacency RIB In as described in RFC4271
+type AdjRIBIn struct {
```

Comment:
This looks like a 1:1 copy of the adjRIBIn we have already. What is stiopping us from using the other?

Reply:
It's not, it's heavily stripped down, and any BGP specific bit is removed as GRP intentionally does not now about ASN and BGP Path attributes. That's also the reason why we can't just use the existing one.